src/pages/en/guides/content-collections.mdx

```diff
@@ -396,6 +396,122 @@ const { Content } = await entry.render();
+
+1. [Upgrade](/en/guides/upgrade-to/v2/) to Astro v2.0 or later and [set up TypeScript](/en/guides/content-collections/#setting-up-typescript) for content collections.
+2. Move Markdown and MDX pages from `src/pages` to subdirectories of `src/content`. Each subdirectory represents a content type. For example, to migrate the [blog posts in the tutorial](/en/tutorial/2-pages/2/), move the contents of `src/pages/posts` to `src/content/posts`. 
+3. Create a `src/content/config.ts` file and [define a schema](/en/guides/content-collections/#defining-a-collection-schema) for each content type. For the blog, we only have one content type, `posts`:
+
+```ts title="src/content/config.ts"
+// Import utilities from `astro:content`
+import { z, defineCollection } from "astro:content";
+// Define a schema for each collection you'd like to validate.
+const postsCollection = defineCollection({
+    schema: z.object({
+      title: z.string(),
+      pubDate: z.date(),
+      description: z.string(),
+      author: z.string(),
+      image: z.object({
+        url: z.string(),
+        alt: z.string()
+      }),
+      tags: z.array(z.string())
+    })
+ });
+// Export a single `collections` object to register your collection(s)
+export const collections = {
+  posts: postsCollection,
+};
+```
+
+:::tip
+If your editor doesn't recognize `astro:content`, make sure you're on the latest version of Astro and try restarting the dev server. 
+:::
+
+4. To [generate routes](/en/guides/content-collections/#generating-routes-from-content), create a `.astro` page with a [dynamic route](/en/core-concepts/routing/#dynamic-routes), then query and render your content. Our tutorial blog uses the static output mode, so we'll use `getStaticPaths` to create a route for each post:
+
+```astro title="[...slug].astro"
+---
+import { getCollection } from 'astro:content';
+import MarkdownPostLayout from '../../layouts/MarkdownPostLayout.astro';
+
+export async function getStaticPaths() {
+  const blogEntries = await getCollection('posts');
+  return blogEntries.map(entry => ({
+    params: { slug: entry.slug }, props: { entry },
+  }));
+}
+
+const { entry } = Astro.props;
+const { Content } = await entry.render();
+---
+<MarkdownPostLayout frontmatter={entry.data}>
+  <Content />
+</MarkdownPostLayout>
+```
+
+Note that we wrapped the content in a layout in the dynamic route. 
+This lets us remove the `layout` definition in each post's frontmatter:
+```md title="src/content/post-1.md" del={2}
+---
+layout: ../../layouts/MarkdownPostLayout.astro
+title: 'My First Blog Post'
+pubDate: 2022-07-01
+...
+---
+```
+
+5. To fetch your content, use [`getCollection`](/en/reference/api-reference/#getcollection) or [`getEntryBySlug`](/en/reference/api-reference/#getentrybyslug) instead of `Astro.glob` or `import` statements. For example, our unique tags logic becomes:
```

Comment:
```suggestion
5. Replace `Astro.glob()` with [`getCollection`](/en/reference/api-reference/#getcollection) to fetch content and metadata from your Markdown files. You will also need to update references to the returned data object, since you will now find your frontmatter values in `data.frontmatter`.

For example, our unique tags logic becomes:
```
1. I think this puts the instruction clearer, up front. I might also hit that this is on "other pages/components or in other files" you might have around your project. e.g. a blog index page, a dynamic page for generating a page for each blog post tag, rss feed. This sets us up nicely for where we're headed re: updating a few different files.

2. I'd also include some kind of reference to `data.frontmatter` here, above the example. (Not in love with wording, just spewed quickly to get it out!)

3. I also think they might not even be using individual file imports anywhere, so I'd hit getCollection (and maybe include getEntryBySlug? but if so, not interwoven with the one we KNOW they will need).

> Similarly, any individual Markdown or MDX file imports should be replaced with [`getEntryBySlug`](/en/reference/api-reference/#getentrybyslug).

THIS could become the "after the example" note, instead of the one about `data.frontmatter`

Reply:
Thanks for all of your comments - should be good to go!